src/api/objects/MutableDomainObject.js

```diff
@@ -113,6 +113,9 @@ class MutableDomainObject {
     }
     $destroy() {
         this._observers.forEach(observer => observer());
+        //TODO: the following deletes will not work since they're not configurable properties.
```

Comment:
I don't follow. What do you mean by configurable properties?

Reply:
The Object.defineProperties API has a configurable option which defaults to false. IF we want to allow deletion of a property like _observers, we need to explicitly set this option to true.

`configurable
    true if the type of this property descriptor may be changed and if the property may be deleted from the corresponding object.
    Defaults to false. `

Reply:
The delete does nothing.

Reply:
Got it, interesting. `this._observers` doesn't need to be deleted, clearing the array would have the same effect. In principle we don't need to delete the reference to `this._globalEventEmitter` either, so let's just drop that line. When the object gets GC'd so does the reference to the global event emitter, so long as it doesn't still have any listeners registered, which it shouldn't.

I think I was just being overcautious when I wrote this.

Reply:
Agreed. I would like to pop the items off this._observers here rather than deleting the property. This is why Ihad suggested pop(). Which way would you prefer?

Reply:
Done.